clients/interface.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2021 The Kubernetes authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clients
+
+import (
+	"context"
+	"github.com/openyurtio/device-controller/api/v1alpha1"
+)
+
+// Options represents the additional parameters required to manipulate the actual object
+type Options map[string]string
+
+// DeviceInterface defines the interfaces which used to create, delete, update, get and list Device objects on edge-side platform
+type DeviceInterface interface {
+	DevicePropertyInterface
+	Create(ctx context.Context, device *v1alpha1.Device, options Options) (*v1alpha1.Device, error)
+	Delete(ctx context.Context, name string, option Options) error
+	Update(ctx context.Context, device *v1alpha1.Device, options Options) (*v1alpha1.Device, error)
+	Get(ctx context.Context, name string, options Options) (*v1alpha1.Device, error)
+	List(ctx context.Context, options Options) ([]v1alpha1.Device, error)
```

Comment:
Since this interface is to get all the device objects of the edge platform, not the devices in k8s, the TypeMeta and ListMeta fields in `v1alpha1.DeviceList` cannot be set, so the []device used here.